docs/site/Project-layout-reference.md

```diff
@@ -0,0 +1,123 @@
+---
+lang: en
+title: 'Project Layout Reference'
+keywords: LoopBack 4.0, LoopBack 4, Node.js, TypeScript
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Project-layout-reference.html
+---
+
+LoopBack 4 project files and directories are in the _application root
+directory_. Within this directory the standard LoopBack 4 project structure has
+these sub-directories:
+
+- `src` - TypeScript source code and configuration files.
+- `public` - Client side assets (JavaScript, HTML, and CSS files) for the home
+  page.
+
+{% include tip.html content="By LoopBack naming conventions, artifacts such as models are grouped under their sub-directories. Also, with LoopBack artifact generators, they create class names in CamelCase and file names are in snake-case. For example, if you create a model named `MyUserModel` with the model generator, it creates files `my-user.model.ts` under `src/models`. See [Naming Convention](Command-line-interface.md#naming-convention) for more information.
+" %}
+
+The following table uses files in the
+[`Todo` example](https://github.com/strongloop/loopback-next/tree/master/examples/todo)
+as code references.
+
+<table style="font-size: 90%;">
+  <thead>
+    <tr>
+      <th width="200">File or directory</th>
+      <th>Description</th>
+      <th width="180">How to access in code</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th colspan="3" style="text-align: center; background-color: #bfbfbf;">Top-level application directory</th>
+    </tr>
+    <tr>
+      <td><code>dist/</code> directory</td>
+      <td>Contains transpiled JavaScript code generated by <code>npm run build</code>.</td>
+      <td>npm run build</td>
```

Comment:
We should mention `npm run build` does incremental build. A full build can be achieved using `npm run clean && npm run build`.